### ISSUE TITLE
i get negative softmax outputs and higher than 1 values !!

### ISSUE THREAD

**Issue by abdallahinf** [Tencent/ncnn#1817]
<img width="909" alt="piiiiii" src="https://user-images.githubusercontent.com/11073061/83882242-2180f100-a742-11ea-9086-f00f28a876d1.png">

i have trained my own model in tensorflow and i have converted it to onnx then to ncnn !!
what is wrong with it ?

**Comment by abdinf** [comment]
@nihui

**Comment by nihui** [comment]
A potential driver issue is discovered recently, which can cause softmax producing negative or very large values on gpu fp16p/fp16s.
workaround in https://github.com/Tencent/ncnn/commit/2345ab7c9268fc6e3f19138735e01d76307e9663

**Comment by nihui** [comment]
针对onnx模型转换的各种问题，推荐使用最新的pnnx工具转换到ncnn
In view of various problems in onnx model conversion, it is recommended to use the latest pnnx tool to convert your model to ncnn

```shell
pip install pnnx
pnnx model.onnx inputshape=[1,3,224,224]
```

详细参考文档
Detailed reference documentation
https://github.com/pnnx/pnnx
https://github.com/Tencent/ncnn/wiki/use-ncnn-with-pytorch-or-onnx#how-to-use-pnnx
